### PR TITLE
Update to Alchemy 7.0

### PR DESCRIPTION
**Pull request by sascha-karnatz**
- Update to Alchemy 7.0
- Update to Rails 7.0
- Remove support for essences and contents
- Use the multisearch configuration on the supported ingredients
- Fix the search for image caption (previously only the object string was stored)
- improve readability of the ingredient extension spec
- add public and upload path to gitignore - the specs are creating small images to test the picture ingredient

### CODE REVIEW

File: app/extensions/alchemy/pg_search/ingredient_extension.rb [hunk 1]
```diff
@@ -17,4 +22,8 @@ def searchable?
   end
 end
 
+# add the PgSearch model to all ingredients
 Alchemy::Ingredient.prepend(Alchemy::PgSearch::IngredientExtension)
+
+# # only enable the search for Text, Richtext, and Picture
```
Comment (tvdeyen) [inline app/extensions/alchemy/pg_search/ingredient_extension.rb]: The comment seems to be wrong

File: spec/dummy/config/initializers/new_framework_defaults_7_0.rb [hunk 2]
```diff
@@ -0,0 +1,148 @@
...
+
+# Active Storage `has_many_attached` relationships will default to replacing the current collection instead of appending to it.
+# Thus, to support submitting an empty collection, the `file_field` helper will render an hidden field `include_hidden` by default when `multiple_file_field_include_hidden` is set to `true`.
+# See https://guides.rubyonrails.org/configuring.html#config-active-storage-multiple-file-field-include-hidden for more information.
+# Rails.application.config.active_storage.multiple_file_field_include_hidden = true
+
+# ** Please read carefully, this must be configured in config/application.rb (NOT this file) **
+# Disables the deprecated #to_s override in some Ruby core classes
+# See https://guides.rubyonrails.org/configuring.html#config-active-support-disable-to-s-conversion for more information.
+# config.active_support.disable_to_s_conversion = true
```
Comment (tvdeyen) [inline spec/dummy/config/initializers/new_framework_defaults_7_0.rb]: The whole file is not necessary.

File: spec/lib/search_spec.rb [hunk 3]
```diff
@@ -198,6 +198,13 @@
         end
 
         it 'should find only the second page' do
+          # fix the first call for ActiveRecord 7.0
+          # result.first would lead to a find on the id - column which is not present in the order - statement,
+          # which than lead to an Postgres - Grouping error
+          #
+          # For now I leave it that way to fix the test and my future self should find a better solution, if this
+          # is really an issue.
```
Comment (tvdeyen) [inline spec/lib/search_spec.rb]: We could try to use

```rb
result.take
```

This returns the first result without a particular order. Returning what the database thinks is first.

File: README.md [hunk 4]
```diff
@@ -76,7 +76,7 @@ The same works for `ingredients` as well
 # elements.yml
 - name: secrets
   ingredients:
-    - name: passwords
+    - role: passwords
```
Comment (tvdeyen) [inline README.md:79]: interesting. when has this change been made? is there a way to amend this?

File: lib/alchemy/pg_search/search.rb [hunk 5]
```diff
@@ -19,18 +19,14 @@ def self.remove_page(page)
...
         page.update_pg_search_document
-        page.all_elements.includes(:ingredients, contents: :essence).find_each do |element|
-          element.contents.each do |content|
-            content.essence.update_pg_search_document if Alchemy::PgSearch.is_searchable?(content.essence_type)
-          end
-
+        page.all_elements.includes(:ingredients).find_each do |element|
           element.ingredients.each do |ingredient|
             ingredient.update_pg_search_document if Alchemy::PgSearch.is_searchable?(ingredient.type)
           end
```
Comment (tvdeyen) [inline lib/alchemy/pg_search/search.rb]: To optimize this we could filter the ingredients before loading all of them

```suggestion
element.ingredients.select { |i| Alchemy::PgSearch.is_searchable?(i.type) }.each do |ingredient|
  ingredient.update_pg_search_document
end
```

**Review by tvdeyen — COMMENTED** [review]
Nice work.

File: README.md (hunk 4 above)
Comment (sascha-karnatz) [reply README.md:79]: This was an old commit in the past: https://github.com/AlchemyCMS/alchemy-pg_search/commit/5be29b5296f49e4c17d8bb65160dac556f2ee53f

**Review by tvdeyen — APPROVED** [review]
Nice!

**Pull request merged**